### ISSUE TITLE
Transient JPA fields being stored

### ISSUE THREAD

**Issue by prabhalar** [Impetus/kundera#567]
Trying out Kundera for Cassandra -noticed that Kundera driver, instead of ignoring the transient field in JPA, stores the information in database.

**Comment by mevivs** [comment]
using thrift client factory or ds java driver?

**Comment by prabhalar** [comment]
Thrift client factory

**Comment by kkmishra** [comment]
version of Kundera you are using ?

**Comment by prabhalar** [comment]
2.11

**Comment by kkmishra** [comment]
I have tried following with 2.11 and working fine for me.

Entity

```
@Entity
@Table(name = "UserInfo", schema = "CompositeCassandra@composite_pu")
public class UserInfo
{

    @Id
    @Column(name = "userInfo_id")
    private String userInfoId;

    @Column(name = "first_name")
    private String firstName;

    @Column(name = "last_name")
    private String lastName;

    @Column(name = "age")
    private int age;

    // Height in cm.
    @Column(name = "height")
    private transient int height;

// getters and setters
```

CRUD

```
       // persist userinfo object.
        UserInfo userInfo = new UserInfo("mevivs_info", "Vivek", "Mishra", 31, 168);
        em.persist(userInfo);

        em.clear();
       // find userinfo object.
        UserInfo foundUser = em.find(UserInfo.class, userInfo.getUserInfoId());
        Assert.assertNotNull(foundUser);
        Assert.assertEquals("Mishra", foundUser.getLastName());
        Assert.assertEquals("Vivek", foundUser.getFirstName());
        Assert.assertEquals(31, foundUser.getAge());
        Assert.assertEquals(0, foundUser.getHeight()); // because 0 is default value of int.  

       // remove userinfo object.
        em.remove(foundUser);

        em.clear();

        UserInfo deletedUser = em.find(UserInfo.class, userInfo.getUserInfoId());
        Assert.assertNull(deletedUser);
```

Please share your scenario, that would be quite helpful.

Thanks
Kuldeep

**Comment by mevivs** [comment]
Are we good to close on this?

-Vivek

**Comment by prabhalar** [comment]
Apologies for not getting back to you for so long. We had the issue with an embeddable object (@Embeddable) and unfortunately I don't have the source code to share with you. Perhaps you can check the above code with any embeddable object and see if it works. I will also look into it once again and get back to you if there any any more updates from my side. If not, feel free to close this issue. Thank  you very much for your help.

**Comment by kkmishra** [comment]
This is what I Verified with @Embeddable as well and working fine i.e. transient fields not persisting.

User Entity :

```
@Entity
@Table(name = "UserInfo", schema = "KunderaTests@addCassandra")
public class UserInfo
{

    @Id
    @Column(name = "userInfo_id")
    private String userInfoId;

    @Column(name = "first_name")
    private String firstName;

    @Column(name = "last_name")
    private String lastName;

    @Column(name = "age")
    private int age;

    // Height in cm.
    @Column(name = "height")
    private transient int height;

    @Embedded
    private EmbeddedInfo info;
// getters and setters
```

Embeddable Entity

```
@Embeddable
public class EmbeddedInfo
{

    @Column(name = "emailId")
    private String emailId;

    @Column(name = "website")
    private String website;

    @Column(name = "noofProfileViewer")
    private transient int noofProfileViewer;
// getters and setters.
```

Test : 

```
@Test
    public void test()
    {
        EntityManagerFactory emf = Persistence.createEntityManagerFactory("addCassandra");
        EntityManager em = emf.createEntityManager();

        // persist userinfo object.
        UserInfo userInfo = new UserInfo();
        userInfo.setAge(25);
        userInfo.setFirstName("Kuldeep");
        userInfo.setHeight(168);
        userInfo.setLastName("Mishra");
        userInfo.setUserInfoId("1");

        EmbeddedInfo info = new EmbeddedInfo();
        info.setEmailId("kuldeep.mishra@gmail.com");
        info.setWebsite("kuldeep.wordpress.com");
        info.setNoofProfileViewer(256);

        userInfo.setInfo(info);
        em.persist(userInfo);

        em.clear();
        // find userinfo object.
        UserInfo foundUser = em.find(UserInfo.class, userInfo.getUserInfoId());
        Assert.assertNotNull(foundUser);
        Assert.assertEquals("Mishra", foundUser.getLastName());
        Assert.assertEquals("Kuldeep", foundUser.getFirstName());
        Assert.assertEquals(25, foundUser.getAge());
        Assert.assertEquals(0, foundUser.getHeight()); // because 0 is default
                                                       // value of int.
        Assert.assertNotNull(foundUser.getInfo());
        Assert.assertEquals("kuldeep.mishra@gmail.com", foundUser.getInfo().getEmailId());
        Assert.assertEquals("kuldeep.wordpress.com", foundUser.getInfo().getWebsite());
        Assert.assertEquals(0, foundUser.getInfo().getNoofProfileViewer());

        // remove userinfo object.
        em.remove(foundUser);

        em.clear();

        UserInfo deletedUser = em.find(UserInfo.class, userInfo.getUserInfoId());
        Assert.assertNull(deletedUser);
    }
```

If you could share your sample code then it would be good for me to reproduce the problem.

-kuldeep

**Comment by prabhalar** [comment]
The last field in the userinfo class should be transient, something like

  @Embedded
@Transient
    private EmbeddedInfo info;
// getters and setters

**Comment by chhavigangwal** [comment]
Fixed in Kundera-2.11.1
